org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/model/JDIDebugTarget.java

```diff
@@ -338,6 +338,10 @@ public class JDIDebugTarget extends JDIDebugElement implements
 	 */
 	private final Map<Long, String> objectLabels = new HashMap<>();
 
+	/**
+	 * Preference storage for active debug session.
+	 */
+	public static final Map<String, Object> debugSessionVariables = new ConcurrentHashMap<>();
```

Comment:
Wjy is this a **static** map and why it is a map at all? If with the the "session" the whole Eclipse session meant, no map is needed, a Boolean flag will be enough. If with the "session" a specific debug session meant, the code is plain wrong as it would affect all other debug sessions runnijg in parallel or after the current one.

Reply:
This can be used to store pref values only for current debug session. other pref values can also be added.

Reply:
other pref values in the sense other features preferences value only for current debug session in future if it requires.

Reply:
> This can be used to store pref values only for current debug session

Then a static map is wrong. Try to start two sessions in parallel.

Reply:
> other pref values in the sense other features preferences value only for current debug session in future if it requires.

If future requires, the code can be refactored.

Reply:
will remove static then.

Reply:
> > other pref values in the sense other features preferences value only for current debug session in future if it requires.
> 
> If future requires, the code can be refactored.

will replace it with boolean flag as you suggested since its more appropriate.

Reply:
I wasnt able to check without static but now I have modified it further to make it more debug sessions specific.
- Tested with parallel debug launches